app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -64,17 +63,16 @@ class ConversationViewModel @Inject constructor(
         }
     }
 
-    fun onMessageChanged(message: TextFieldValue) {
+    fun onMessageChanged(message: String) {
         conversationViewState = conversationViewState.copy(messageText = message)
     }
 
     fun sendMessage() {
         viewModelScope.launch {
-            val messageText = conversationViewState.messageText
             //TODO what if conversationId is null???
-            sendTextMessage(conversationId!!, messageText.text)
-            conversationViewState = conversationViewState.copy(messageText = messageText.copy(""))
+            sendTextMessage(conversationId!!, conversationViewState.messageText)
         }
+        conversationViewState = conversationViewState.copy(messageText = "")
```

Comment:
If the message sending action fails we still want to remove the text in the input field?

Reply:
actually good question, I do not think it was in the designs or we did not think about yet, I think in general we should give more thought to how to handle errors and progress on this screen

Reply:
after sending the message we would like to clear the message input text field, do not know if that is what you mean ?

Reply:
I meant that if the message send action fails, then why should we also clear the input field? Shouldn't we check if the result was succesful before doing so?

Reply:
the way it works on facebook for example is that the textfield is cleared but when the message sending fails the failure status is shown on the message itself, think this is a question to Wolfgang :P

Reply:
ok, not a blocker for now then, but maybe something to revisit once the whole chat funcionality is up and running